echo_server.py

```diff
@@ -1,25 +1,30 @@
 import socket
 
-server_socket = socket.socket(
-    socket.AF_INET,
-    socket.SOCK_STREAM,
-    socket.IPPROTO_IP)
-server_socket.bind(('127.0.0.1', 50000))
-server_socket.listen(10)
-i = 4  # the exact number of tests we have
-while i > 0:
-    conn, addr = server_socket.accept()
-    buffsize = 16
-    msg = ""
-    done = False
-    while not done:
-        part = conn.recv(buffsize)
-        print part
-        if len(part) < buffsize:
-            done = True
-        msg = "{}{}".format(msg, part)
-    out = "{}{}".format("I heard: ", msg)
-    conn.sendall(out)
-    conn.close()
-    i -= 1
-server_socket.close()
+
+def echo_server():
+    server_socket = socket.socket(
+        socket.AF_INET,
+        socket.SOCK_STREAM,
+        socket.IPPROTO_IP)
+    server_socket.bind(('127.0.0.1', 50000))
+    server_socket.listen(10)
+    try:
+        while True:
+            conn, addr = server_socket.accept()
+            buffsize = 16
+            msg = ""
+            done = False
+            while not done:
+                part = conn.recv(buffsize)
+                if len(part) < buffsize:
+                    done = True
+                msg = "{}{}".format(msg, part)
+            out = "{}{}".format("I heard: ", msg)
+            conn.sendall(out)
```

Comment:
The only problem here is that instead of echoing *exactly* what the client sent, you are altering it before sending it back.  That contradicts the assignment.